sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_create_test_live.cpp

```diff
@@ -228,6 +230,87 @@ TEST_F(KeyVaultKeyClient, CreateRsaHsmKey)
+    output[2 * i + 1] = hexMap[input[i] & 0x0F];
+  }
+
+  return output;
+}
+
+TEST_F(KeyVaultKeyClient, ReleaseKey)
+{
+  auto const keyName = GetTestName() + "2";
+  auto const& client = GetClientForTest(keyName);
+
+  Azure::Security::Attestation::AttestationClient attestationClient(
+      "https://sharedwus.wus.attest.azure.net");
+
+  MaaSandboxClient sandboxClient("https://maasandboxsvc.azurewebsites.net");
+  attestationClient.RetrieveResponseValidationCollateral();
+
+  auto rsaKey = CreateRsaKeyOptions("rsaKeyName");
+  rsaKey.KeySize = 2048;
+  rsaKey.ExpiresOn = std::chrono::system_clock::now() + std::chrono::hours(24 * 365);
+
+  client.CreateRsaKey(rsaKey);
+
+  KeyVaultKey cloudRsaKey = client.GetKey("rsaKeyName").Value;
+  cloudRsaKey.Properties.Id = "rsaKeyName";
+  Azure::Core::Json::_internal::json destJson;
+  Azure::Core::Json::_internal::json destJson2;
+  Azure::Security::KeyVault::Keys::_detail::JsonWebKeySerializer::JsonWebKeySerialize(
+      cloudRsaKey.Key, destJson2);
+  destJson["keys"].emplace_back(destJson2);
+  auto dest = destJson.dump();
+  std::string result = Base64Url::Base64UrlEncode(std::vector<uint8_t>(dest.begin(), dest.end()));
+  auto generatedToken = sandboxClient.GenerateQuote(AttestationDataType::Json, result);
+  attestationClient.RetrieveResponseValidationCollateral();
+
+  auto attestResponse = attestationClient.AttestOpenEnclave(
+      generatedToken.Value,
+      AttestOptions{AttestationData{
+          std::vector<uint8_t>(dest.begin(), dest.end()), AttestationDataType::Binary}});
+
+  Azure::Security::KeyVault::Keys::CreateKeyOptions options;
+  options.KeyOperations.push_back(Azure::Security::KeyVault::Keys::KeyOperation::Sign);
+  options.KeyOperations.push_back(Azure::Security::KeyVault::Keys::KeyOperation::Verify);
+  options.ReleasePolicy = KeyReleasePolicy();
+  options.ReleasePolicy.Value().Immutable = false;
+  // cspell:disable
+  std::string dataStr = "{"
+                        "\"anyOf\" : [ {"
+                        "\"allOf\" : [{\"claim\" : \"x-ms-sgx-mrsigner\", \"equals\" : \""
+      + BinaryToHexString(attestResponse.Value.Body.SgxMrSigner.Value())
+      + "\""
+        "} ],"
+        "\"authority\" : \"https://sharedwus.wus.attest.azure.net\""
+        "} ],"
+        " \"version\" : \"1.0.0\""
+        "} ";
+  // cspell:enable
+  auto jsonParser = json::parse(dataStr);
+  options.ReleasePolicy.Value().Data = jsonParser.dump();
+  options.Exportable = true;
+  auto keyResponse
+      = client.CreateKey(keyName, Azure::Security::KeyVault::Keys::KeyVaultKeyType::EcHsm, options);
+
+  KeyReleaseOptions relOpt;
+  relOpt.Target = attestResponse.Value.RawToken;
+  relOpt.Encryption = KeyEncryptionAlgorithm::RSA_AES_KEY_WRAP_256;
+  auto result2 = client.ReleaseKey(keyName, keyResponse.Value.Properties.Version, relOpt);
+  EXPECT_NE(result2.Value.Value.length(), size_t(0));
+  EXPECT_EQ(result2.RawResponse->GetStatusCode(), HttpStatusCode::Ok);
```

Comment:
If you want to show the end-to-end key release process, you could also use the key you got back in line 266 to decrypt the released key :).

Reply:
sure, no idea how though

Reply:
I believe you thake the response from MHSM and pass it into the Key.Decrypt API which will give you the actual key which was released. But I'm not 100% sure.

Reply:
it's a jwt thingy , not an encrypted string